public/components/common/hooks/use-query.ts

```diff
@@ -45,5 +45,12 @@ export function useQuery(): [
 }
 
 export const useQueryManager = () => {
-  return useState(getDataPlugin().query.queryString.getQuery());
-};
+  const [query, setQuery] = useState(getDataPlugin().query.queryString.getQuery());
+  useEffect(() => {
+    const { unsubscribe } = getDataPlugin().query.queryString.getUpdates$().subscribe((q) => {
+      setQuery(q);
+    });
+    return () => unsubscribe();
```

Comment:
We must remember to check whether this unsubscription works correctly before merging